api/routes/auth.js

```diff
@@ -0,0 +1,105 @@
+const { SECRET_KEY } = process.env
+const router = require('express').Router()
+const bcrypt = require('bcrypt')
+const jsonwebtoken = require('jsonwebtoken')
+const User = require('../models/user')
+
+router.post('/signup', async (req, res, next) => {
+  const status = 200
+  try {
+    const { username, password } = req.body
+    const user = await User.findOne({ username })
+    if (user) throw new Error (`User already exists.`)
+    
+    //Create User, if user does not yet exist.
+    const saltrounds = 12
+    const hashPass = await bcrypt.hash(password, saltrounds)
+    const response = await User.create({username, password: hashPass})
+
+    res.status(status).json({ status, response })
+
+  } catch (e) {
+    console.error(e)
+
+    const error = new Error('User already exists or does not meet validation requirements.')
+    error.status = 422
+    next(error)
+
+    res.status(status).json({ status, message })
+
+  }  
+
+})
+
+router.post('/login', async (req, res, next) => { 
+
+  try {
+    const { username, password } = req.body
+    const user = await User.findOne({ username })
+    if (!user) throw new Error('Username could not be found.')
+      
+    const validPassword = await bcrypt.compare(password, user.password)
+    if (!validPassword) throw new Error('Password is incorrect.')
```

Comment:
You'll want to give the same errors in either case. Otherwise, it's easier for someone to "guess" at a valid username.